File: srcs/app/lib/models/task.dart
class Task {
  final String id;
  final String title;
  final String status;
  final String? assignedAgentId;
  final List<String> dependencies;

  const Task({
    required this.id,
    required this.title,
    required this.status,
    this.assignedAgentId,
    this.dependencies = const [],
  });

  factory Task.fromJson(Map<String, dynamic> json) {
    return Task(
      id: json['id'] as String? ?? '',
      title: json['title'] as String? ?? '',
      status: json['status'] as String? ?? 'PENDING',
      assignedAgentId: json['assignedAgentId'] as String? ?? json['assigned_agent_id'] as String?,
      dependencies: (json['dependencies'] as List<dynamic>?)?.map((e) => e as String).toList() ?? [],
    );
  }
}

File: src/server/models/task.go
package models

import "time"

// Task represents a task in the shared queue.
type Task struct {
	ID              string
	MissionID       string
	ParentPlanID    string
	Dependencies    []string
	Title           string
	Description     string
	Priority        string
	Status          string // PENDING, READY, IN_PROGRESS, COMPLETED, BLOCKED, FAILED
	AssignedAgentID string
	Payload         string
	CreatedAt       time.Time
	UpdatedAt       time.Time

	ActionRisk      string   `json:"action_risk,omitempty"`
	ApprovalStatus  string   `json:"approval_status,omitempty"`
	ProposedContent string   `json:"proposed_content,omitempty"`
}

// TaskDependency represents a dependency relationship between tasks.
type TaskDependency struct {
	TaskID          string
	DependsOnTaskID string
}

// SharedTask is used primarily by TaskManager for multi-tenant queue operations.
type SharedTask struct {
	ID              string   `json:"id"`
	OrganizationID  string   `json:"organization_id"`
	ParentPlanID    string   `json:"parent_plan_id"`
	Dependencies    []string `json:"dependencies"`
	Title           string   `json:"title"`
	Description     string   `json:"description,omitempty"`
	Status          string   `json:"status"`
	AssignedAgentID string   `json:"assigned_agent_id,omitempty"`
	Priority        string   `json:"priority"`
	Payload         string   `json:"payload"`
	LockedUntil     *time.Time `json:"locked_until,omitempty"`
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`

	ActionRisk      string   `json:"action_risk,omitempty"`
	ApprovalStatus  string   `json:"approval_status,omitempty"`
	ProposedContent string   `json:"proposed_content,omitempty"`
}
